xpk.py

```diff
@@ -1964,9 +1969,48 @@ def create_vertex_tensorboard(args) -> map:
                                             tensorboard_name=tensorboard_name)
   if instance_id:
     xpk_print(f'Tensorboard instance {tensorboard_name} is successfully created.')
-    tensorboard_config['location'] = args.tensorboard_region
-    tensorboard_config['instance_name'] = tensorboard_name
-    tensorboard_config['instance_id'] = instance_id
+    tensorboard_config['tensorboard_location'] = args.tensorboard_region
+    tensorboard_config['tensorboard_name'] = tensorboard_name
+    tensorboard_config['tensorboard_id'] = instance_id
+  return tensorboard_config
+
+
+def create_vertex_experiment(args) -> map:
+  """Creates an Experiment in Vertex AI.
+
+  Args:
+    args: user provided arguments.
+
+  Returns:
+    map containing Vertex Tensorboard configurations.
+  """
+  metadata_configmap_name = f'{args.cluster}-{_CLUSTER_METADATA_CONFIGMAP}'
+  cluster_config_map = get_cluster_configmap(args, metadata_configmap_name)
+
+  if cluster_config_map is None or 'tensorboard_name' not in cluster_config_map:
+    xpk_print('No Vertex Tensorboard instance has been created in cluster create.'
+              ' Run `xpk cluster create --create-vertex-tensorboard` before running'
+              ' `xpk workload create --use-vertex-tensorboard` to create a Vertex'
+              ' Tensorboard instance.')
+    xpk_exit(1)
+
+  tensorboard_config = {}
+  tensorboard_config['tensorboard_project'] = args.project
+  tensorboard_config['tensorboard_location'] = cluster_config_map['tensorboard_location']
+  tensorboard_config['tensorboard_name'] = cluster_config_map['tensorboard_name']
+  experiment_name = args.experiment_name
+  if experiment_name is None:
+    experiment_name = f'{args.cluster}-{args.workload}'
```

Comment:
Just curious how the experiment_name is used in tensorboard for my background info for me?

Reply:
In Vertex, we create a Tensorboard instance and then we can attach multiple experiments to a single TB instance. So, whenever a user schedules a workload, we will create an experiment and attach it to a TB instance created for that cluster.

In order to upload logs to an experiment created for a workload that points to a Tensorboard instance in Vertex, `experiment_name` would be required. This is the API we use: https://cloud.google.com/vertex-ai/docs/experiments/tensorboard-upload-existing-logs